src/Symfony/Bundle/WebProfilerBundle/Twig/WebProfilerExtension.php

```diff
@@ -24,19 +26,42 @@ class WebProfilerExtension extends \Twig_Extension
      * @var ValueExporter
      */
     private $valueExporter;
+    /**
+     * @var HtmlDumper
+     */
+    private $dumper;
+
+    public function __construct(HtmlDumper $dumper = null)
+    {
+        $this->dumper = $dumper ?: new HtmlDumper();
+    }
 
     /**
      * {@inheritdoc}
      */
     public function getFunctions()
     {
         return array(
-            new \Twig_SimpleFunction('profiler_dump', array($this, 'dumpValue')),
+            new \Twig_SimpleFunction('profiler_dump', array($this, 'dumpValue'), array('is_safe' => array('html'))),
```

Comment:
This looks dangerous to me: the legacy implementation is still is use, but it's now considered "html-safe".
What about using a new function instead?